packages/chat-stateful-client/src/ChatContext.ts

```diff
@@ -48,14 +48,15 @@ export class ChatContext {
   private _fullsizeImageQueue: ResourceDownloadQueue | undefined = undefined;
   constructor(
     maxListeners?: number,
-    /* @conditional-compile-remove(teams-inline-images-and-file-sharing) */ credential?: CommunicationTokenCredential
+    /* @conditional-compile-remove(teams-inline-images-and-file-sharing) */ credential?: CommunicationTokenCredential,
+    /* @conditional-compile-remove(teams-inline-images-and-file-sharing) */ endpoint?: string
   ) {
     this._logger = createClientLogger('communication-react:chat-context');
     this._emitter = new EventEmitter();
     /* @conditional-compile-remove(teams-inline-images-and-file-sharing) */
     if (credential) {
-      this._inlineImageQueue = new ResourceDownloadQueue(this, credential);
-      this._fullsizeImageQueue = new ResourceDownloadQueue(this, credential);
+      this._inlineImageQueue = new ResourceDownloadQueue(this, { credential, endpoint: endpoint ?? '' });
+      this._fullsizeImageQueue = new ResourceDownloadQueue(this, { credential, endpoint: endpoint ?? '' });
```

Comment:
in what case would `endpoint` be undefined?

Reply:
This is a good question. I think the implementation and arguments can be cleaned up a bit, but technically the credential and the endpoint in this case should go hand-in-hand. There is no point in passing the credential without the endpoint, and no point in passing the endpoint without the credential (so maybe they should be put into a one object).

Reply:
Is this not a question of source compatibility with previous API versions? If this is non-optional, it breaks existing callers of this public API.

Reply:
The constructor takes an optional endpoint, as it has to be optional to maintain source compatibility.

We can either propagate this optional or define it to be non-optional like this.